### PR TITLE
added first testing infrastructure and rest api to fcm

### PR DESCRIPTION
**Pull request by bendavidelad**

### CODE REVIEW

File: src/main/java/com/whistleBlower/demo/buisness_logic/BusinessLogic.java [hunk 1]
```diff
@@ -25,13 +30,60 @@ public BusinessLogic() {
...
+        Group group = message.getGroup();
+        List<User> users = group.getUsers();
+        users.forEach(user -> sendMessageToUser(user, message));
+    }
+
+    public List<Group> getGroups(){
+        return groups;
+    }
+
+    private void sendGroupToUser(Group group, User user){ // TODO finish this function
```
Comment (orwatt) [inline src/main/java/com/whistleBlower/demo/buisness_logic/BusinessLogic.java:57]: You have TODO comment, Are you done with it?

File: src/main/java/com/whistleBlower/demo/buisness_logic/BusinessLogic.java [hunk 2]
```diff
@@ -25,13 +30,60 @@ public BusinessLogic() {
...
+        headers.add("Authorization", "key=IAFUIASXOASCNI#R*U&FDJAS(");
+        HttpEntity<Group> request = new HttpEntity<>(group, headers);
+
+        RestTemplate restTemplate = new RestTemplate();
+        String result = restTemplate.postForObject(uri, request, String.class);
+    }
 
-}
+
+    private void sendMessageToUser(User user, Message message){  //  TODO finish this function
```
Comment (orwatt) [inline src/main/java/com/whistleBlower/demo/buisness_logic/BusinessLogic.java:70]: You have TODO comment, Are you done with it?

File: src/main/java/com/whistleBlower/demo/buisness_logic/BusinessLogic.java [hunk 3]
```diff
@@ -25,13 +30,60 @@ public BusinessLogic() {
...
+        final String uri = "http://localhost:8027/";
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Content-Type", "application/json");
+        headers.add("Authorization", "key=IAFUIASXOASCNI#R*U&FDJAS(");
+
```
Comment (orwatt) [inline src/main/java/com/whistleBlower/demo/buisness_logic/BusinessLogic.java:76]: Key and directory should be constants

**Review by orwatt — CHANGES_REQUESTED** [review]

**Review by orwatt — APPROVED** [review]
Good Job 👍

**Review by orwatt — APPROVED** [review]

**Review by elihorovitz — APPROVED** [review]

**Pull request merged**